conda/plugins/types.py

```diff
@@ -347,3 +348,20 @@ class CondaRequestHeader:
 
     name: str
     value: str
+
+
+@dataclass
+class CondaPostTransaction:
+    """
+    Return type to use when defining a post-transaction hook.
+
+    For details on how this is used, see
+    :meth:`~conda.plugins.hookspec.CondaSpecs.conda_post_transactions`.
+
+    :param name: Post transaction name (this is just a label).
+    :param run: Function to run after each transaction; accepts a single parameter which
+        is the action that has just been executed
+    """
+
+    name: str
+    run: Callable[[_Action], None]
```

Comment:
Should the plugin declare on which `_Action` instances it should run? This would allow us to skip a lot of calls. Otherwise, the plugin itself needs to check the `_Action` type every time, possibly just to ignore it. If it should run for everything, then it could just `_Action`. In other words:

```suggestion
    run: Callable[[_Action], None]
    action_type: _Action = _Action
```